packages/interactivity/src/init.ts

```diff
@@ -29,6 +29,38 @@ export const initialVdom = new WeakMap< Element, ComponentChild[] >();
 
 // Initialize the router with the initial DOM.
 export const init = async () => {
+	const pendingNodes = new Set();
+
+	const intersectionObserver = new window.IntersectionObserver(
+		async ( entries ) => {
+			for ( const entry of entries ) {
+				if ( ! entry.isIntersecting ) {
+					continue;
+				}
+
+				const node = entry.target;
+				intersectionObserver.unobserve( node );
+				pendingNodes.delete( node );
+				if ( pendingNodes.size === 0 ) {
+					intersectionObserver.disconnect();
+				}
+
+				if ( ! hydratedIslands.has( node ) ) {
+					const fragment = getRegionRootFragment( node );
+					const vdom = toVdom( node );
+					await splitTask();
+					hydrate( vdom, fragment );
+					await splitTask();
```

Comment:
Why does this yield at the end of the process? In the original code that happened before, right above the `getRegionRootFragment` call.

Reply:
As I recall, I moved it here because given that this is now inside of an `IntersectionObserver` callback, it is already running as part of a separate task. Therefore, it doesn't make sense to `splitTask` at the beginning of the loop since then essentially there would be two yieldings to the main thread back to back. So this is why I moved the `await splitTask()` calls further down here.